ogre/src/OgreLidarVisual.cc

```diff
@@ -0,0 +1,306 @@
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->rayStrips)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->rayLines)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->deadZoneRayFans)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  this->ClearPoints();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Init()
+{
+  BaseLidarVisual::Init();
+  this->Create();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Create()
+{
+  this->ClearPoints();
+  this->dataPtr->receivedData = false;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::ClearPoints()
+{
+  this->dataPtr->lidarPoints.clear();
+  this->dataPtr->receivedData = false;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::ClearVisualData()
+{
+  this->dataPtr->noHitRayStrips.clear();
+  this->dataPtr->deadZoneRayFans.clear();
+  this->dataPtr->rayLines.clear();
+  this->dataPtr->rayStrips.clear();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::SetPoints(const std::vector<double> &_points)
+{
+  this->dataPtr->lidarPoints = _points;
+  this->dataPtr->receivedData = true;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Update()
+{
+  if (!this->dataPtr->receivedData || this->dataPtr->lidarPoints.size() == 0)
+  {
+    ignwarn << "New lidar data not received. Exiting update function"
+            << std::endl;
+    return;
+  }
+
+  this->dataPtr->receivedData = false;
+  double horizontalAngle = this->minHorizontalAngle;
+  double verticalAngle = this->minVerticalAngle;
+
+  if (this->horizontalCount > 1)
+  {
+    this->horizontalAngleStep =
+        (this->maxHorizontalAngle - this->minHorizontalAngle) /
+              (this->horizontalCount - 1);
+  }
+  if (this->verticalCount > 1)
+  {
+    this->verticalAngleStep =
+        (this->maxVerticalAngle - this->minVerticalAngle) /
+              (this->verticalCount - 1);
+  }
+
+  if (this->dataPtr->lidarPoints.size() !=
+                  this->verticalCount * this->horizontalCount)
+  {
+    ignwarn << "Size of lidar data inconsistent with rays."
+            << " Exiting update function."
+            << std::endl;
+    return;
+  }
+
+  // Process each point from received data
+  // Every line segment, and every triangle is saved separately,
+  // as a pointer to a DynamicLine
+  for (unsigned int j = 0; j < this->verticalCount; ++j)
+  {
+    horizontalAngle = this->minHorizontalAngle;
+
+    if (j+1 > this->dataPtr->rayLines.size())
+    {
+      // Ray Strips fill in between the line areas that intersect an object
+      std::shared_ptr<OgreDynamicLines> line =
+                  std::shared_ptr<OgreDynamicLines>(
+                              new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+      line->setMaterial("Lidar/BlueStrips");
```

Comment:
windows build is failing due to the `setMaterial` calls in this function as windows is using a newer version of ogre. Looks like ogre deprecated this API in version 1.10.8 in favor of the overloaded `setMaterial(MaterialPtr &)` function, see [here](https://github.com/OGRECave/ogre/blob/v1.10.8/OgreMain/include/OgreSimpleRenderable.h#L77). Can you add a check for OGRE_VERSION and use the string version if version is <= 1.10.8? here's an example: https://github.com/ignitionrobotics/ign-rendering/blob/master/ogre/src/OgreRenderEngine.cc#L99